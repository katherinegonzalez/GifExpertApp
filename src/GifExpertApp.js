import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball Z'];

    const [categories, setCategories] = useState(['One Punch']);

    /*const handleAdd = () => {
        // setCategories([...categories, 'HunterXHunter']); ó también se puede:
        setCategories(cats => [...cats, 'HunterXHunter']);
    }*/

    return(
        <>
            <h2>GifExpertApp</h2>
            <hr />
            {/* <button onClick={()=> handleAdd()}>Agregar</button> */}
            <AddCategory setCategories={setCategories} />
            <ol>
                { 
                    categories.map(category =>
                        <GifGrid 
                            key={category}
                            category={category} />
                        )
                }
            </ol>
        </>
    );
}

export default GifExpertApp;