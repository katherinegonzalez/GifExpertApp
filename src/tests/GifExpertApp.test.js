import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
    
    test('Debe motrar <GifExpertApp /> correctamente', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories} /> );
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
})
