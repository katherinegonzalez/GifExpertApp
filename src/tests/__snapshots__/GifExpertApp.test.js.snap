// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe motrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
